src/pages/index.js

```diff
@@ -4,78 +4,224 @@ import FormValidator from "../components/FormValidator.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import UserInfo from "../components/UserInfo.js";
-import { initialCards, config } from "../utils/constants.js";
+import { config } from "../utils/constants.js";
 import "./index.css";
+import PopupWithConfirmation from "../components/PopupWithConfirmation";
+import Api from "../components/Api.js";
+// import { setLoading } from "../components/PopupWithForm.js";
+
+//APIs
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "4a5b23f0-f2a7-4209-a8e7-d3bcf73a20e6",
+    "Content-Type": "application/json",
+  },
+});
+
+export function renderCard(cardData) {
+  const card = new Card(
+    cardData,
+    "#card-template",
+    handleCardClick,
+    handleDeleteCard,
+    handleLikeCard
+  );
+  return card.getView();
+}
+
+let userInfo;
+let cardSection;
+
+// Load initial cards
+api.loadPage()
+  .then(([initialCards, userData]) => {
+    
+    // Initialize and render the card section
+    cardSection = new Section(
+      {
+        items: initialCards,
+        renderer: (cardData) => {
+          const cardElement = renderCard(cardData)
+          cardSection.addItem(cardElement);
+        }
+      },
+      "#cards__list"
+    );
+    cardSection.renderItems();
+
+    // Update user profile
+    userInfo = new UserInfo({
+      nameSelector: ".profile__title",
+      jobSelector: ".profile__description",
+      imageSelector: ".profile__image"
+    });
+    userInfo.setUserInfo(userData);
+  })
+  .catch((error) => {
+    console.log(error);
+  });
+
+// Preview image
+function handleCardClick(name, link) {
+  previewImageModal.open(name, link);
+}
+
+// Add card function
+function handleAddCardSubmit(values) {
+  const name = values.title;
+  const link = values.link;
+    
+  // Set loading state
+  addCardPopup.setLoading(true);
+
+  // Call API method to add a new card
+  api.addNewCard(name, link)
+    .then((newCard) => {
+      // Render the newly created card
+      const cardElement = renderCard(newCard);
+
+      addCardPopup.setLoading(false);
```

Comment:
`      addCardPopup.setLoading(false);
`

You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case: `success` or `failure`.

**Please, correct it everywhere**